yt_dlp/extractor/ninenow.py

```diff
@@ -59,26 +55,35 @@ def _real_extract(self, url):
             cache = page_data.get(kind, {}).get('%sCache' % kind, {})
             if not cache:
                 continue
-            common_data = (cache.get(current_key) or list(cache.values())[0])[kind]
+            common_data = {
+                'episode': (cache.get(current_key) or list(cache.values())[0])[kind],
+                'season': (cache.get(current_key) or list(cache.values())[0]).get('season', None)
+            }
             break
         else:
             raise ExtractorError('Unable to find video data')
 
-        video_data = common_data['video']
+        # Video Data extraction
+        if not self.get_param('allow_unplayable_formats') and try_get(common_data, lambda x: x['episode']['video']['drm'], bool):
+            self.report_drm(display_id)
+        brightcove_id = try_get(
+            common_data,
+            lambda x: x['episode']['video']['brightcoveId'] or f"ref:{x['episode']['video']['referenceId']}",
+            compat_str)
+        video_id = try_get(common_data, lambda x: x['episode']['video']['id'] or brightcove_id, compat_str)
```

Comment:
don't think this works

`id` appears to be an int. `or brightcove_id` will not be reached if `id` key doesn't exist, so it should be outside the try_get

```suggestion
        video_id = str_or_none(try_get(common_data, lambda x: x['episode']['video']['id'])) or brightcove_id
```

I'm not sure if its a good idea having different possible video ids for the same video, might mess with archive files if one was to fail extraction. But it was like that before so we should probably keep it? @pukkandan 

Also might need to update your tests for this.

Reply:
Yeah, fallbacks for id is generally bad. But since it was existing code, maybe some videos are known not to have `video_data['id']`. Better not touch it unless we know for sure